boards/posix/native_posix/native_posix_defconfig

```diff
@@ -4,6 +4,13 @@ CONFIG_SOC_POSIX=y
 CONFIG_BOARD_NATIVE_POSIX_32BIT=y
 CONFIG_CONSOLE=y
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=1000000
+
+# For native posix k_sleep is used in the data path as well as for shell input
+# detection, hence data processing is at least two ticks per packet. To support
+# 5ms ISO interval bidirectional data the system shall never stall for more
+# than 1 ms.
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
```

Comment:
This does not look like the change you would do in the default board configuration, but something you would do for your application..
Is there any reason to not do it in your application configuration file or application board configuration overlay?

Reply:
As it is a general limiter for how many BT packets/events the native_posix build can handle per second, I see no reason not to make this change apply to all native_posix builds. It took quite a while for me to debug the low data rate, hence I'm sure it will save time for others in the future - it will also make the native_posix build behave more like a real system.

The implementation of userchan will be able to handle multiple packets if multiple packets are queued up in the BlueZ socket buffer. But as soon as you have emptied the socket buffer the system will sleep for one or two ticks before you read again. With 10 ms ticks this is more than two LC3 frames, hence will give an audio-gap for low latency audio. Also some controllers have a limited number of RX buffers for ISO data, hence will start to discard data.
As you say, the intention for this build is to have easy interaction with real host peripherals (such as audio), hence I think this change is needed, and I don't think I'm doing anything wrong. I think the wrong thing here is the way the native_posix implementation is done, where calls to read do not yield the task, which is why sleep is used in the first place - to busy-poll for data.
One of the next steps for ISO data is to include LC3 processing of ISO data, where you would need to process one packet every 7.5ms. The easiest way to actually get audio-data into Zephyr at the moment (where the audio subsystem is not in place) is to use a Linux build and pipe raw audio data in/out using a socket or file - which is why I would like to have the ISO data flow stable in the native posix build.

Reply:
> I see no reason not to make this change apply to all native_posix builds.

native_posix is relatively rarely used with the BT host. It is not compiled in "real time mode" in most cases either.

> it will also make the native_posix build behave more like a real system

That was never the intention ;-). Native posix was born as a faster than real time test setup.

Anyhow, if you want to have this kconfig option default to a higher tick rate when BT_USER_CHAN is selected, that would be another matter (but @jhedberg would need to be ok with that).

>where calls to read do not yield the task

Call to the host read do not yield the task, because they are direct calls to the host OS, and have nothing to do with the Zephyr OS. There has been for a while a request to provide an API which can do what you would like #6044. But that is certainly not trivial to implement.

Reply:
I see the tests fail due to 1000 ticks per second - hence I'll need to reduce to 500 or make it based on USERCHAN and/or the BT shell application.

Reply:
>native_posix is relatively rarely used with the BT host.
I'll move this to the native_posix.conf overlay file for the relevant application then.